opengrok-indexer/src/main/java/org/opengrok/indexer/history/BazaarHistoryParser.java

```diff
@@ -110,79 +111,81 @@ public void processStream(InputStream input) throws IOException {
+                            } catch (ParseException e) {
+                                //
+                                // Overriding processStream() thus need to comply with the
+                                // set of exceptions it can throw.
+                                //
+                                throw new IOException("Failed to parse history timestamp:" + s, e);
+                            }
+                            ++state;
+                        }
+                        break;
+                    case 3:
+                        // Expect the commit message to follow immediately after
+                        // the timestamp, and that everything up to the list of
+                        // modified, added and removed files is part of the commit
+                        // message.
+                        if (s.startsWith("modified:") || s.startsWith("added:") || s.startsWith("removed:")) {
+                            ++state;
+                        } else if (s.startsWith("  ")) {
+                            // Commit messages returned by bzr log -v are prefixed
+                            // with two blanks.
+                            entry.appendMessage(s.substring(2));
                         }
-                    }
-                    break;
-                default:
-                    LOGGER.log(Level.WARNING, "Unknown parser state: {0}", state);
-                    break;
+                        break;
+                    case 4:
+                        // Finally, store the list of modified, added and removed
+                        // files. (Except the labels.)
+                        if (!(s.startsWith("modified:") || s.startsWith("added:") || s.startsWith("removed:"))) {
+                            // The list of files is prefixed with blanks.
+                            s = s.trim();
+
+                            int idx = s.indexOf(" => ");
+                            if (idx != -1) {
+                                s = s.substring(idx + 4);
+                            }
+
+                            File f = new File(myDir, s);
+                            try {
+                                String name = env.getPathRelativeToSourceRoot(f);
+                                entry.addFile(name.intern());
+                            } catch (ForbiddenSymlinkException e) {
+                                LOGGER.log(Level.FINER, e.getMessage());
+                                // ignored
+                            } catch (InvalidPathException e) {
+                                LOGGER.log(Level.WARNING, e.getMessage());
+                            }
+                        }
+                        break;
+                    default:
+                        LOGGER.log(Level.WARNING, "Unknown parser state: {0}", state);
+                        break;
                 }
+            }
         }
 
         if (entry != null && state > 2) {
```

Comment:
style-wise, I'd expect `Objects.nonNull()` to be also used here, esp. given this is the same variable.

Reply:
Found a way to avoid this null check